src/qtism/data/results/Context.php

```diff
@@ -182,4 +187,25 @@ public function hasSessionIdentifiers(): bool
     {
         return (bool)$this->sessionIdentifiers->count();
     }
+
+    /**
+     * Check if the context has a session identifier with the given sourceId.
+     *
+     * @param $sourceId
+     * @return bool
+     */
+    private function hasSessionIdentifierWithSourceId($sourceId): bool
+    {
+        if (!$this->hasSessionIdentifiers()) {
```

Comment:
not required as in this case the foreach loop would be skipped and we would still return false